pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -265,14 +265,24 @@ func (r *Reconciler) delete() error {
 		klog.Infof("Machine %v not found during delete, skipping", r.machine.Name)
 		return nil
 	}
-	zone := r.providerSpec.Zone
-	operation, err := r.computeService.InstancesDelete(r.projectID, zone, r.machine.Name)
+	// The core machine controller will not call us back on error
+	// or an explicit request to re-queue by returning
+	// RequeueAfterError.
+	//
+	// By the time delete is called the machine object has gone.
+	// If were to call machine.Update() and add an annotation for
+	// the operation.Id (so that we could keep track of it in
+	// follow-up calls) the Update() will fail.
+	//
+	// Ideally we would like to return nil when the cloud
+	// operation is "DONE". However, the current design of the
+	// core controller does not allow for an idempotent delete
+	// operation.
+	operation, err := r.computeService.InstancesDelete(r.projectID, r.providerSpec.Zone, r.machine.Name)
 	if err != nil {
-		return fmt.Errorf("failed to delete instance via compute service: %v", err)
-	}
-	if op, err := r.waitUntilOperationCompleted(zone, operation.Name); err != nil {
-		return fmt.Errorf("failed to wait for delete operation via compute service. Operation status: %v. Error: %v", op, err)
+		return fmt.Errorf("failed to delete instance for machine %q via compute service: %v", machineID(r.machine), err)
 	}
+	klog.Infof("Delete Operation #%v initiated for machine %q", operation.Id, machineID(r.machine))
```

Comment:
I'm unsure of how it might happen, but would the delete operation potentially not take effect for some reason?  If so, then we will have removed the machine-record, we'll have no indication later that an instance is left hanging around.  Maybe we should always fail here, and just let Exists() handle it in a later run.